packages/transformers/webext/src/WebExtTransformer.js

```diff
@@ -0,0 +1,273 @@
+  ['browser_action', 'theme_actions', 'light'],
+  ['browser_action', 'theme_actions', 'dark'],
+  ['page_action', 'default_icon'],
+  ['page_action', 'default_popup'],
+  ['background', 'scripts'],
+  ['chrome_url_overrides'],
+  ['devtools_page'],
+  ['options_ui', 'page'],
+  ['sidebar_action', 'default_icon'],
+  ['sidebar_action', 'default_panel'],
+  ['storage', 'managed_schema'],
+  ['theme', 'images', 'theme_frame'],
+  ['theme', 'images', 'additional_backgrounds'],
+  ['user_scripts', 'api_script'],
+];
+
+async function collectDependencies(
+  asset: MutableAsset,
+  program: any,
+  ptrs: {[key: string]: any, ...},
+  hot: boolean,
+) {
+  // isEntry used whenever strictly necessary to preserve filename
+  // also for globs because it's wasteful to write out every file name
+  const fs = asset.fs;
+  const filePath = asset.filePath;
+  if (program.default_locale) {
+    const locales = join(dirname(filePath), '_locales');
+    let err = !(await fs.exists(locales))
+      ? 'key'
+      : !(await fs.exists(join(locales, program.default_locale)))
+      ? 'value'
+      : null;
+    if (err) {
+      throw new ThrowableDiagnostic({
+        diagnostic: [
+          {
+            message: 'Invalid Web Extension manifest',
+            origin: '@parcel/transformer-webext',
+            filePath,
+            codeFrame: {
+              codeHighlights: [
+                {
+                  ...getJSONSourceLocation(ptrs['/default_locale'], err),
+                  message: `Localization directory${
+                    err == 'value' ? ' for ' + program.default_locale : ''
+                  } does not exist: ${relative(
+                    dirname(filePath),
+                    join(locales, program.default_locale),
+                  )}`,
+                },
+              ],
+            },
+          },
+        ],
+      });
+    }
+    for (const locale of await fs.readdir(locales)) {
+      asset.addURLDependency(posix.join('_locales', locale, 'messages.json'), {
+        isEntry: true,
+        pipeline: 'url',
+      });
+    }
+  }
+  let needRuntimeBG = false;
+  if (program.content_scripts) {
+    for (let i = 0; i < program.content_scripts.length; ++i) {
+      const sc = program.content_scripts[i];
+      for (const k of ['css', 'js']) {
+        const assets = sc[k] || [];
+        for (let j = 0; j < assets.length; ++j) {
+          assets[j] = asset.addURLDependency(assets[j], {
+            isEntry: true,
+            loc: {
+              filePath,
+              ...getJSONSourceLocation(
+                ptrs[`/content_scripts/${i}/${k}/${j}`],
+                'value',
+              ),
+            },
+          });
+        }
+      }
+      if (hot && sc.js && sc.js.length) {
+        needRuntimeBG = ptrs[`/content_scripts/${i}/js/${sc.js.length - 1}`];
+        sc.js.push(
+          asset.addURLDependency(
+            relative(dirname(asset.filePath), RUNTIME),
+            {},
+          ),
```

Comment:
Use `resolveFrom` instead of the relative path: https://github.com/parcel-bundler/parcel/blob/aff1426c88f465cdea1a977c4b6633417e61eb69/packages/transformers/js/src/visitors/modules.js#L287-L295